### PR TITLE
Add jobs for CAPI release-1.0

### PR DESCRIPTION
**Pull request by anusha94**
- add periodics-release-1-0 jobs
- add periodics-release-1-0-upgrade jobs
- add presubmits-release-1-0 jobs
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/5410

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-0.yaml [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+  - name: pull-cluster-api-build-release-1-0
+    decorate: true
+    path_alias: sigs.k8s.io/cluster-api
+    always_run: true
+    labels:
+      preset-service-account: "true"
+    branches:
+    # The script this job runs is not in all branches.
+    - ^release-1.0$
+    - ^operator-1.0$
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-release-1-0.yaml:12]:
```suggestion
```
You can drop this branch across the file. There is only an `operator-0.4` branch and it looks like the operator work will be moved to a separate repo.

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
+periodics:
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0.yaml:1]: Can you please also add a copy of `periodic-cluster-api-e2e-v1alpha3-v1beta1-upgrade-main` (from cluster-api-periodics-main.yaml)

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0-upgrades.yaml [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+    repo: cluster-api
+    base_ref: release-1.0
+    path_alias: sigs.k8s.io/cluster-api
+  - org: kubernetes
+    repo: kubernetes
+    base_ref: master
+    path_alias: k8s.io/kubernetes
+  spec:
+    containers:
+    - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20211022-00550799a0-1.21
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0-upgrades.yaml:20]:
```suggestion
    - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20211022-00550799a0-1.22
```
Please upgrade to 1.22 in all release-1.0 files (the idea is that we use the same version which we use in CAPI, CAPI v1.0.0 is using controller-runtime 0.10 which is using Kubernetes 1.22)

**Review by sbueringer — COMMENTED** [review]
A few nits, otherwise looks good!

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0.yaml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+      - "./scripts/ci-test.sh"
+      resources:
+        requests:
+          cpu: 7300m
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-cluster-api-1.0
+    testgrid-tab-name: capi-test-release-1-0
+    testgrid-alert-email: sig-cluster-lifecycle-cluster-api-alerts@kubernetes.io
+    testgrid-num-failures-to-alert: "2"
+- name: periodic-cluster-api-e2e-v1alpha3-v1beta1-upgrade-main
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0.yaml:25]:
```suggestion
- name: periodic-cluster-api-e2e-v1alpha3-v1beta1-upgrade-release-1-0
```

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0.yaml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+  interval: 1h
+  decorate: true
+  labels:
+    preset-dind-enabled: "true"
+    preset-kind-volume-mounts: "true"
+  extra_refs:
+  - org: kubernetes-sigs
+    repo: cluster-api
+    base_ref: main
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0.yaml:34]:
```suggestion
    base_ref: release-1.0
```

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0.yaml [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+
+        # we need privileged mode in order to do docker in docker
+        securityContext:
+          privileged: true
+        resources:
+          requests:
+            cpu: 7300m
+  annotations:
+    testgrid-dashboards: sig-cluster-lifecycle-cluster-api
+    testgrid-tab-name: capi-e2e-v1alpha3-v1beta1-upgrade-main
```
Comment (sbueringer) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-release-1-0.yaml:64]:
```suggestion
    testgrid-dashboards: sig-cluster-lifecycle-cluster-api-1.0
    testgrid-tab-name: capi-e2e-v1alpha3-v1beta1-upgrade-release-1-0
```

**Review by sbueringer — COMMENTED** [review]
Last round of nits, feel free to squash directly

**Comment by sbueringer** [comment]
Thank you, looks great!

/lgtm

/assign @fabriziopandini

**Comment by fabriziopandini** [comment]
@anusha94 thanks for this PR, great stuff!

/approve
/lgtm

**Pull request merged**